core/trino-main/src/main/java/io/trino/server/ServerMainModule.java

```diff
@@ -406,6 +408,10 @@ protected void setup(Binder binder)
         binder.bind(TypeRegistry.class).in(Scopes.SINGLETON);
         binder.bind(TypeManager.class).to(InternalTypeManager.class).in(Scopes.SINGLETON);
         newSetBinder(binder, Type.class);
+        binder.bind(TypeRegistrations.class).asEagerSingleton();
+
+        // function
```

Comment:
this comment is not helpful. Group `FunctionRegistrations` and `TypeRegistrations` and explain why we need those as eager singletons (hopefully we can get rid of those - but I am not sure yet).

Reply:
refactored